server/connection.go

```diff
@@ -119,6 +120,11 @@ func (conn *Connection) close() {
 		conn.statusLock.Unlock()
 		return
 	}
+
+	if conn.heartbeatTimer != nil {
```

Comment:
Not critical, but Lock - Unlock is `critical section` and should be much faster, can u move Stop after unlock?